src/test-utils/browser-automation/browser-driver.js

```diff
@@ -93,26 +85,50 @@ export default class BrowserDriver {
   stopBrowser() {
     return Promise.resolve()
       .then(_ => this.page.waitFor(1000))
-      .then(_ => this.browser.close())
-      .catch(error => {
-        console.error(addColor(ERR_AUTOMATION, COLOR.BRIGHT_RED)); // eslint-disable-line
-        throw error;
-      });
+      .then(_ => this.browser.close());
   }
 
-  startServer(config = {}) {
-    this.child = this.execFile(
-      config.process || DEFAULT_CONFIG.process,
-      config.parameters || DEFAULT_CONFIG.parameters,
-      config.options || DEFAULT_CONFIG.options,
-      (err, stdout) => {
-        if (err) {
-          this.console.error(err);
-          return;
+  startServer(config = {}, maxRetryTimes = 30) {
+    const newConfig = Object.assign({}, DEFAULT_CONFIG, config);
+    return new Promise((resolve, reject) => {
+      log.log({
+        message: `Binding to port: ${newConfig.port}`,
+        color: COLOR.YELLOW
+      })();
+      const timeout = setTimeout(() => { // eslint-disable-line
+        resolve();
+      }, 2000);
+      this.child = this.execFile(
+        newConfig.process,
+        [...newConfig.parameters, '--port', `${newConfig.port}`],
+        newConfig.options,
+        error => {
+          if (error) {
+            clearTimeout(timeout); // eslint-disable-line
+            log.log({
+              message: `Failed to bind port: ${newConfig.port}`,
+              color: COLOR.YELLOW
+            })();
+            reject(error);
+          }
+        }
+      );
+    })
+      .then(() => {
+        this.port = newConfig.port;
+      })
+      .catch(error => {
+        if (maxRetryTimes > 0) {
+          newConfig.port++;
+          return this.startServer(newConfig, maxRetryTimes - 1);
+        } else { // eslint-disable-line
+          log.log({
+            message: 'Failed to start server, use \'killall node\' to stop existing services',
+            color: COLOR.RED
+          })();
+          throw error;
```

Comment:
Does this actually display the real error? If it wasn't related to ports it would be good to show it.